Mudul Crash

Das Modul stürzt ab, wenn keine Abfahrtzeiten (mehr) gefunden werden - z.B. beim Beginn der Betriebspause. Die Anzeige bleibt dann mit nur einer Zeile bei der letzten Abfahrt und der Zeit "Now" hängen bis der MagicMirror neu gestartet wird.

Klar, kein Problem.

Ich kann mit Einträgen in der `/home/pi/.pm2/logs/mm-error.log` nicht wirklich etwas anfangen, aber hilft dir das evtl?
```
[2021-03-18 23:49:49.919] [ERROR]  Whoops! There was an uncaught exception...
[2021-03-18 23:49:49.923] [ERROR]  TypeError: Cannot read property 'length' of undefined
    at Class.processTrains (/home/pi/MagicMirror/modules/MMM-HH-LocalTransport/node_helper.js:120:49)
    at /home/pi/MagicMirror/modules/MMM-HH-LocalTransport/node_helper.js:100:30
    at Request.handleRequestResponse [as _callback] (/home/pi/MagicMirror/modules/MMM-HH-LocalTransport/node_modules/unirest/index.js:552:70)
    at Request.self.callback (/home/pi/MagicMirror/modules/MMM-HH-LocalTransport/node_modules/request/request.js:185:22)
    at Request.emit (events.js:200:13)
    at Request.<anonymous> (/home/pi/MagicMirror/modules/MMM-HH-LocalTransport/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:200:13)
    at IncomingMessage.<anonymous> (/home/pi/MagicMirror/modules/MMM-HH-LocalTransport/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:288:20)
    at IncomingMessage.emit (events.js:205:15)
[2021-03-18 23:49:49.952] [ERROR]  MagicMirror will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
[2021-03-18 23:49:49.955] [ERROR]  If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues
```

--- a/node_helper.js
+++ b/node_helper.js
@@ -117,24 +117,26 @@ module.exports = NodeHelper.create({
         var datetime = new Date(this.parseDate(data.time.date));
         datetime.setHours(hours);
         datetime.setMinutes(mins);
-        for (var i = 0, count = data.departures.length; i < count; i++) {
-            var trains = data.departures[i];
-            if (trains.timeOffset < 0) {
-                continue;
+        if (typeof data.departures !== "undefined") {
+            for (var i = 0, count = data.departures.length; i < count; i++) {
+                var trains = data.departures[i];
+                if (trains.timeOffset < 0) {
+                    continue;
+                }
+                var departureTime = datetime.setMinutes(datetime.getMinutes() + trains.timeOffset);
+                if (trains.delay != 0 && trains.delay != undefined) {
+                    delay = trains.delay;
+                } else {
+                    delay = null;
+                }
+                this.trains.push({
+                    departureTimestamp: trains.timeOffset,
+                    delay: delay,
+                    name: trains.line.name,
+                    id: trains.line.id,
+                    to: trains.line.direction
+                });
             }
-            var departureTime = datetime.setMinutes(datetime.getMinutes() + trains.timeOffset);
-            if (trains.delay != 0 && trains.delay != undefined) {
-                delay = trains.delay;
-            } else {
-                delay = null;
-            }
-            this.trains.push({
-                departureTimestamp: trains.timeOffset,
-                delay: delay,
-                name: trains.line.name,
-                id: trains.line.id,
-                to: trains.line.direction
-            });
         }
         this.loaded = true;
         this.sendSocketNotification("TRAINS", this.trains);

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "MMM-HH-LocalTransport",
-  "version": "0.9.2",
+  "version": "0.9.3",
   "description": "Local transport in Hamburg module for MagicMirror",
   "repository": {
     "type": "git",

